src/main/java/com/khoros/twitterapp/TwitterAuthorization.java

```diff
@@ -0,0 +1,59 @@
+package com.khoros.twitterapp;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.hibernate.validator.constraints.NotBlank;
+import twitter4j.conf.*;
+
+public class TwitterAuthorization {
+    @NotBlank
+    private CharSequence consumerKey;
```

Comment:
why CharSequence?

Reply:
Because NotBlank validation checks that it has at least 1 character and ignores whitespace. That's the only reason. For the specialized validation that seems to match the variable.